src/pages/iou/SplitBillDetailsPage.tsx

```diff
@@ -26,46 +24,25 @@ import * as IOU from '@userActions/IOU';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
 import type SCREENS from '@src/SCREENS';
-import type {PersonalDetailsList, Report, ReportAction, Session, Transaction} from '@src/types/onyx';
 import type {Participant} from '@src/types/onyx/IOU';
-import type {ReportActions} from '@src/types/onyx/ReportAction';
 import {isEmptyObject} from '@src/types/utils/EmptyObject';
 
-type SplitBillDetailsPageTransactionOnyxProps = {
-    /** The current transaction */
-    transaction: OnyxEntry<Transaction>;
+type SplitBillDetailsPageProps = WithReportAndReportActionOrNotFoundProps & StackScreenProps<SplitDetailsNavigatorParamList, typeof SCREENS.SPLIT_DETAILS.ROOT>;
 
-    /** The draft transaction that holds data to be persisited on the current transaction */
-    draftTransaction: OnyxEntry<Transaction>;
-};
-
-type SplitBillDetailsPageOnyxPropsWithoutTransaction = {
-    /** The personal details of the person who is logged in */
-    personalDetails: OnyxEntry<PersonalDetailsList>;
-
-    /** The active report */
-    report: OnyxEntry<Report>;
-
-    /** Array of report actions for this report */
-    reportActions: OnyxEntry<ReportActions>;
-
-    /** Session info for the currently logged in user. */
-    session: OnyxEntry<Session>;
-};
-
-type SplitBillDetailsPageOnyxProps = SplitBillDetailsPageTransactionOnyxProps & SplitBillDetailsPageOnyxPropsWithoutTransaction;
-
-type SplitBillDetailsPageProps = WithReportAndReportActionOrNotFoundProps &
-    SplitBillDetailsPageOnyxProps &
-    StackScreenProps<SplitDetailsNavigatorParamList, typeof SCREENS.SPLIT_DETAILS.ROOT>;
-
-function SplitBillDetailsPage({personalDetails, report, route, reportActions, transaction, draftTransaction, session}: SplitBillDetailsPageProps) {
+function SplitBillDetailsPage({report, reportAction}: SplitBillDetailsPageProps) {
     const styles = useThemeStyles();
-    const reportID = report?.reportID ?? '-1';
     const {translate} = useLocalize();
     const theme = useTheme();
-    const reportAction = useMemo(() => reportActions?.[route.params.reportActionID] ?? ({} as ReportAction), [reportActions, route.params.reportActionID]);
-    const participantAccountIDs = ReportActionsUtils.isMoneyRequestAction(reportAction) ? ReportActionsUtils.getOriginalMessage(reportAction)?.participantAccountIDs ?? [] : [];
+
+    const reportID = report?.reportID ?? '-1';
+    const originalMessage = reportAction && ReportActionsUtils.isMoneyRequestAction(reportAction) ? ReportActionsUtils.getOriginalMessage(reportAction) : undefined;
+    const IOUTransactionID = originalMessage?.IOUTransactionID ?? '-1';
+    const participantAccountIDs = originalMessage?.participantAccountIDs ?? [];
+
+    const [transaction] = useOnyx(`${ONYXKEYS.COLLECTION.TRANSACTION}${IOUTransactionID}`);
+    const [draftTransaction] = useOnyx(`${ONYXKEYS.COLLECTION.TRANSACTION_DRAFT}${IOUTransactionID}`);
```

Comment:
Hi @hoangzinh, I thought Rory has addressed your comments since he has worked on the changes on SplitBill pages. Sorry if I missed anything. Is there any changes I have to make?

Reply:
This change is causing lint failure. 
```
const IOUTransactionID = originalMessage?.IOUTransactionID || '-1';
```
```
Prefer using nullish coalescing operator (`??`) instead of a logical or (`||`), as it is a safer operator
```
Should I go back to using nullish calescing?